lib/elixir/lib/enum.ex

```diff
@@ -1284,7 +1284,7 @@ defmodule Enum do
       "2 = 4 = 6"
 
   """
-  @spec map_join(t, String.t, (element -> any)) :: String.t
+  @spec map_join(t, String.t, (element -> String.t)) :: String.t
```

Comment:
I am not sure that `String.Chars.t` is a type, and if it was it would have to be `any`, so I think `any` is correct.